gradle-spaghetti-typescript-plugin/src/main/java/com/prezi/spaghetti/typescript/gradle/SpaghettiTypeScriptPlugin.java

```diff
@@ -86,6 +91,18 @@ public void execute(TypeScriptTestBinary testBinary) {
 			}
 		});
 
+		spaghettiExtension.registerDefinitionSearchSourceDirs(new Function<Void, Iterable<File>>() {
+			public Iterable<File> apply(Void input) {
+				Set<TypeScriptSourceSet> sources = typeScriptExtension.getSources().getByName("main").withType(TypeScriptSourceSet.class);
+
+				Collection<Iterable<File>> sourceDirs = new HashSet<Iterable<File>>();
+				for (TypeScriptSourceSet sourceSet : sources) {
+					sourceDirs.add(sourceSet.getSource().getSrcDirs());
+				}
+				return Iterables.concat(sourceDirs);
+			}
+		});
+
```

Comment:
@rosadam This is the best way I found for the spaghetti-typescript-plugin to extend the search dirs in the spaghetti-plugin. It's really verbose and hard to understand, but it works.